visualization/app/codeCharta/ui/attributeSideBar/attributeSideBarPrimaryMetrics/primaryMetrics.selector.ts

```diff
@@ -0,0 +1,44 @@
+import { createSelector } from "../../../state/angular-redux/store"
+import { selectedNodeSelector } from "../../../state/selectors/selectedNode.selector"
+import { CcState } from "../../../state/store/store"
+import { Metric, shouldShowAttributeTypeSelector } from "../util/metricHelper"
+import { primaryMetricNamesSelector } from "./primaryMetricNames.selector"
+
+export type PrimaryMetrics = {
+	area: Metric
+	height: Metric
+	color: Metric
+	edge: { name: string; incoming: number; outgoing: number } | undefined
+}
+
+export const primaryMetricsSelector: (state: CcState) => PrimaryMetrics | undefined = createSelector(
+	[selectedNodeSelector, primaryMetricNamesSelector],
+	(selectedNode, primaryMetricNames) => {
+		if (!selectedNode?.attributes) return
```

Comment:
Is this case ever triggered outside of tests? If not, it's more a kind of assertion, right?

Reply:
According to [defined type here](https://github.com/MaibornWolff/codecharta/blob/main/visualization/app/codeCharta/codeCharta.model.ts#L331) `attributes is always present`. I was afraid due to #2460. But I checked previous implemantation, which doesn't include this check. So good catch :) Nullish check for `atttributes` will be removed with next push